docs/root/faq/extensions/contract.rst

```diff
@@ -0,0 +1,7 @@
+.. _faq_filter_contract:
+
+Is there a contract my HTTP filter must adhere to?
+==================================================
+
+* A filter's ``decodeData`` implementation must not return ``FilterHeadersStatus::ContinueAndEndStream`` when called with ``end_stream`` set.
```

Comment:
There are a bunch more in https://github.com/envoyproxy/envoy/blob/master/include/envoy/http/filter.h that might be worth re-iterating. E.g. the StopIteration contract, which headers you can safely expect to exist in decodeHeaders() (e.g. Path? Host?).

@envoyproxy/maintainers what other contracts exist here?

Reply:
Added notes about:
- headers encoding/decoding
- data encoding/decoding
- trailers encoding/decoding
- known headers: Host and Path (optional for CONNECT requests).
Anything else that comes to mind?